lib/jsdom/living/nodes/Node-impl.js

```diff
@@ -294,23 +312,52 @@ class NodeImpl extends EventTargetImpl {
     }
   }
 
-  removeChild(/* Node */ oldChildImpl) {
+  removeChild(oldChildImpl) {
+    // https://dom.spec.whatwg.org/#concept-node-pre-remove
+
     if (!oldChildImpl || domSymbolTree.parent(oldChildImpl) !== this) {
       throw new DOMException("The object can not be found here.", "NotFoundError");
     }
 
+    // https://dom.spec.whatwg.org/#concept-node-remove
+
     if (this._ownerDocument) {
       this._ownerDocument._runPreRemovingSteps(oldChildImpl);
     }
+
     domSymbolTree.remove(oldChildImpl);
+
+    if (oldChildImpl._assignedSlot) {
+      assignSlotable(oldChildImpl._assignedSlot);
+    }
+
+    let hasSlotDescendant = oldChildImpl._localName === "slot";
+    if (!hasSlotDescendant) {
+      for (const child of domSymbolTree.treeIterator(oldChildImpl)) {
+        if (child._localName === "slot") {
+          hasSlotDescendant = true;
+          break;
+        }
+      }
+    }
+
+    if (hasSlotDescendant) {
+      assignSlotableForTree(getRoot(this));
+
+      // TODO: if the node is removed from is parent at 9., the root will always be itself ?!
+      assignSlotableForTree(getRoot(oldChildImpl));
```

Comment:
In the [remove node algorithm](https://dom.spec.whatwg.org/#concept-node-remove), the root of the node at this point ([12.2](https://dom.spec.whatwg.org/#ref-for-concept-tree-root%E2%91%A1%E2%93%AA)) is always the node itself since it has already been detached.
Am I missing something here?

Reply:
Not sure! Please file a spec bug.

Reply:
https://github.com/whatwg/dom/issues/691